syntax = "proto3";
import "google/protobuf/empty.proto";

service DataManager {
    rpc ChangeConfig (ServiceConfig) returns (google.protobuf.Empty) {}

    rpc GetService (GetServiceRequest) returns (ServiceConfig) {}
}

// The request message containing the user's name.
message ServiceConfig {
    string name = 1;
    string url = 2;
    int32 frequency = 3;
    int32 alerting_window = 4;
    int32 allowed_resp_time = 5;
    string phone_number = 6;
    string email = 7;
}

message GetServiceRequest {
    string name = 1;
}
